File: ProjectsInjava/RecyclerView/app/src/main/java/com/example/recyclerview/MainAdapter.java
package com.example.recyclerview;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;


import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MainAdapter extends RecyclerView.Adapter<MainAdapter.MainViewHolder> {

    private ArrayList<PostItem> posts;

    private ItemClickListener itemClickListener;

    public MainAdapter(ArrayList<PostItem> posts, ItemClickListener itemClickListener) {
        super();
        this.posts = posts;
        this.itemClickListener = itemClickListener;
    }

    @NonNull
    @Override
    public MainViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater
                .from(parent.getContext())
                .inflate(R.layout.item_row_person, parent, false);
        return new MainViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MainViewHolder holder, final int position) {
        holder.bind(posts.get(position), itemClickListener);
    }

    @Override
    public int getItemCount() {
        return posts.size();
    }

    public class MainViewHolder extends RecyclerView.ViewHolder {

        private TextView postName;
        private TextView description;
        private ImageView image;
        private ImageView profileImage;

        public MainViewHolder(@NonNull View itemView) {
            super(itemView);
            postName = itemView.findViewById(R.id.postName);
            description = itemView.findViewById(R.id.description);
            image = itemView.findViewById(R.id.imageView);
            profileImage = itemView.findViewById(R.id.imageView2);
        }

        public void bind(final PostItem postItem, final ItemClickListener itemClickListener) {
            postName.setText(postItem.getPostName());
            description.setText(postItem.getDescription());
            image.setImageResource(R.drawable.pain);
            profileImage.setImageResource(R.drawable.naruto);
            if(itemClickListener != null){
                itemView.setOnClickListener(new View.OnClickListener(){
                    @Override
                    public void onClick(View view) {
                        itemClickListener.onItemClick(postItem);
                    }
                });
            }
        }
    }

    public interface ItemClickListener {
        void onItemClick(PostItem postItem);
    }
}

File: KotlinProjects/RecyclerKotlin/app/src/main/java/com/example/recyclerkotlin/MainAdapter.kt
package com.example.recyclerkotlin

import android.content.Intent

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat.startActivity
import androidx.recyclerview.widget.RecyclerView

class MainAdapter(var posts: List<Post>,val callback: Callback) : RecyclerView.Adapter<MainAdapter.MainHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int)
            = MainHolder(LayoutInflater.from(parent.context).inflate(R.layout.for_post, parent, false))

    override fun getItemCount() = posts.size

    override fun onBindViewHolder(holder: MainHolder, position: Int) {
        holder.bind(posts[position])
        holder.myimg.visibility = View.VISIBLE
    }

    inner class MainHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private val title = itemView.findViewById<TextView>(R.id.postTitle)
        private val description = itemView.findViewById<TextView>(R.id.postDescription)
        val myimg = itemView.findViewById<ImageView>(R.id.imgView_postPic)

        private val share = itemView.findViewById<ImageView>(R.id.shareImg)

        fun bind(post: Post) {
            title.text = post.title
            description.text = post.description.take(40) + "..."

            itemView.setOnClickListener {
                if (adapterPosition != RecyclerView.NO_POSITION) callback.onItemClicked(posts[adapterPosition])
            }

            share.setOnClickListener{
                val shareBody = post.title + "\n" + post.description
                val sharingIntent = Intent(Intent.ACTION_SEND)
                sharingIntent.type = "text/plain"
                sharingIntent.putExtra(Intent.EXTRA_TEXT, shareBody)
               // startActivity(Intent.createChooser(sharingIntent, "send"))

            }


        }
    }

    interface Callback {
        fun onItemClicked(post: Post)
    }

}